ixmp/core.py

```diff
@@ -1230,24 +1259,33 @@ def read_sol_from_gdx(self, path, filename, comment=None,
     def has_solution(self):
         """Return :obj:`True` if the Scenario has been solved.
 
-        If ``has_solution() == True``, model solution data is exists in the
-        database.
+        If ``has_solution() == True``, model solution data exists in the db.
         """
         return self._jobj.hasSolution()
 
-    def remove_solution(self):
-        """Delete the model solution.
+    def remove_solution(self, first_model_year=None):
+        """Delete the model solution (variables and equations) and timeseries
+        data marked as `meta=False` (see :meth:`TimeSeries.add_timeseries`).
+
+        Parameters
+        ----------
+        first_model_year: int, optional
+            If given, timeseries data marked as `meta=False` is removed
+            only for years from `first_model_year` onwards.
 
         Raises
         ------
         ValueError
-            If Scenario has no solution.
+            If Scenario has no solution or if `first_model_year` is not `int`.
         """
         if self.has_solution():
             self.clear_cache()  # reset Python data cache
-            self._jobj.removeSolution()
+            if isyear(first_model_year, 'first_model_year'):
```

Comment:
see my comment whwere `isyear()` is defined

Reply:
done